components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchConfiguration.java

```diff
@@ -0,0 +1,158 @@
+ * limitations under the License.
+ */
+package org.apache.camel.component.elasticsearch5;
+
+import java.util.List;
+
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+import org.apache.camel.spi.UriPath;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+
+@UriParams
+public class ElasticsearchConfiguration {
+
+    private List<InetSocketTransportAddress> transportAddressesList;
+
+    @UriPath @Metadata(required = "true")
+    private String clusterName;
+    @UriParam(enums = "INDEX, UPDATE, BULK, BULK_INDEX, GET_BY_ID, MULTIGET, DELETE, EXISTS, SEARCH, MULTISEARCH, DELETE_INDEX")
+    private String operation;
+    @UriParam
+    private String indexName;
+    @UriParam
+    private String indexType;
+    @UriParam
+    private int waitForActiveShards = ElasticsearchConstants.DEFAULT_FOR_WAIT_ACTIVE_SHARDS;
+    @UriParam
+    private String ip;
+    @UriParam
+    private String transportAddresses;
+    @UriParam
+    private int port = ElasticsearchConstants.DEFAULT_PORT;
+    @UriParam(defaultValue = "true")
+    private Boolean clientTransportSniff = true;
+
+    /**
+     * Name of the cluster
+     */
+    public String getClusterName() {
+        return clusterName;
+    }
+
+    public void setClusterName(String clusterName) {
+        this.clusterName = clusterName;
+    }
+
+    /**
+     * What operation to perform
+     */
+    public String getOperation() {
+        return operation;
+    }
+
+    public void setOperation(String operation) {
+        this.operation = operation;
+    }
+
+    /**
+     * The name of the index to act against
+     */
+    public String getIndexName() {
+        return indexName;
+    }
+
+    public void setIndexName(String indexName) {
+        this.indexName = indexName;
+    }
+
+    /**
+     * The type of the index to act against
+     */
+    public String getIndexType() {
+        return indexType;
+    }
+
+    public void setIndexType(String indexType) {
+        this.indexType = indexType;
+    }
+
+    /**
+     * The TransportClient remote host ip to use
+     */
+    public String getIp() {
+        return ip;
+    }
+
+    public void setIp(String ip) {
+        this.ip = ip;
+    }
+
+    /**
+     * Comma separated list with ip:port formatted remote transport addresses to use.
+     * The ip and port options must be left blank for transportAddresses to be considered instead.
+     */
+    public String getTransportAddresses() {
+        return transportAddresses;
+    }
+
+    public void setTransportAddresses(String transportAddresses) {
+        this.transportAddresses = transportAddresses;
+    }
+
+    /**
+     * The TransportClient remote port to use (defaults to 9300)
+     */
+    public int getPort() {
+        return port;
+    }
+
+    public void setPort(int port) {
+        this.port = port;
+    }
+    
+    /**
```

Comment:
Can we improve the javadoc with a bit more documentation. For example for this you can document the various enums that are possible and what each of them mean/do.